packages/contracts/src/Multisig.sol

```diff
@@ -264,23 +284,32 @@ contract Multisig is
             revert DateOutOfBounds({limit: _startDate, actual: _endDate});
         }
 
-        proposalId = _createProposal({
-            _creator: _msgSender(),
-            _metadata: _metadata,
-            _startDate: _startDate,
-            _endDate: _endDate,
-            _actions: _actions,
-            _allowFailureMap: _allowFailureMap
-        });
+        proposalId = hashProposal(_actions, _metadata);
 
         // Create the proposal
         Proposal storage proposal_ = proposals[proposalId];
 
+        // Multisig doesn't allow `minApprovals` settings to be less than 0.
+        // If it is, that means proposal hasn't been created yet.
+        if (proposal_.parameters.minApprovals != 0) {
+            revert ProposalAlreadyExists(proposalId);
+        }
+
         proposal_.parameters.snapshotBlock = snapshotBlock;
         proposal_.parameters.startDate = _startDate;
         proposal_.parameters.endDate = _endDate;
         proposal_.parameters.minApprovals = multisigSettings.minApprovals;
 
+        TargetConfig memory currentTarget = getTargetConfig();
+
+        if (currentTarget.target == address(0)) {
+            proposal_.target = address(dao());
+            proposal_.operation = Operation.Call;
```

Comment:
Good idea, one question though. How would consumer(in this case, Multisig) would figure out if target is 0 or not in case it needs to know this for some reason ? if we return `dao` when target is 0 inside  `getTargetConfig`, the scenario I explained will not be possible. wdyt ?